### PR TITLE
Move node-template Cargo.toml dependencies to workspace

### PR DESCRIPTION
**Pull request by kalaninja**
Updates `node-template-release` tool to collect git dependencies of node-template modules and puts them into `[workspace.dependencies]` of root `Cargo.toml`.

### CODE REVIEW

File: scripts/ci/node-template-release/src/main.rs [hunk 1]
```diff
@@ -55,10 +64,19 @@ fn find_cargo_tomls(path: PathBuf) -> Vec<PathBuf> {
...
-	let options = CopyOptions::new();
-	dir::copy(node_template, dest_path, &options).expect("Copies node-template to tmp dir");
+/// Parse the given `Cargo.toml`
+fn parse_cargo_toml(file: &Path) -> CargoToml {
+	let mut content = String::new();
+	File::open(file)
+		.expect("Cargo.toml exists")
+		.read_to_string(&mut content)
+		.expect("Reads file");
+	content.parse().expect("Cargo.toml is a valid toml file")
```
Comment (bkchr) [inline scripts/ci/node-template-release/src/main.rs:74]:
```suggestion
	fs::read_to_string(file).expect("Failed to read `Cargo.toml`").parse().expect("Cargo.toml is a valid toml file")
```

File: scripts/ci/node-template-release/src/main.rs [hunk 2]
```diff
@@ -55,10 +64,19 @@ fn find_cargo_tomls(path: PathBuf) -> Vec<PathBuf> {
...
+}
+
+fn write_cargo_toml(path: &Path, cargo_toml: CargoToml) {
+	let mut file = File::create(path).expect(&format!("Creates `{}`.", path.display()));
+	write!(file, "{}", cargo_toml).expect("Writes `Cargo.toml`");
```
Comment (bkchr) [inline scripts/ci/node-template-release/src/main.rs:79]:
```suggestion
	fs::write(path, cargo_toml.to_string()).expect("Writes `Cargo.toml`");
```

**Review by bkchr — APPROVED** [review]

File: scripts/ci/node-template-release/src/main.rs [hunk 3]
```diff
@@ -8,23 +6,28 @@ use std::{
 	process::Command,
 };
 
-use glob;
-
+use clap::Parser;
```
Comment (ggwpez) [inline scripts/ci/node-template-release/src/main.rs:9]: How does this work?  
What is this `node-template-release` being used for? Is it for the developer hub?

Comment (bkchr) [reply scripts/ci/node-template-release/src/main.rs:9]: It rips off the `node-template` from the Substrate repo, creates a workspace Cargo.toml and zips the resulting files.

File: scripts/ci/node-template-release/src/main.rs [hunk 4]
```diff
@@ -76,82 +94,221 @@ fn get_git_commit_id(path: &Path) -> String {
...
+
+	let mut panic_unwind = Table::new();
+	panic_unwind.insert("panic", value("unwind"));
+	let mut profile = Table::new();
+	profile.insert("release", Item::Table(panic_unwind));
+	cargo_toml.insert("profile", Item::Table(profile.into()));
+}
+
+#[test]
+fn test_update_root_cargo_toml() {
```
[the remark this answers is not in the document]
Comment (kalaninja) [reply scripts/ci/node-template-release/src/main.rs]: done

File: scripts/ci/node-template-release/src/main.rs (hunk 2 above)
Comment (kalaninja) [reply scripts/ci/node-template-release/src/main.rs]:
```
fs::write(path, cargo_toml.to_string())
	.unwrap_or_else(|e| panic!("Failed to write `{}`: {}", path.display(), e));
```
to show which file causes the problem

File: scripts/ci/node-template-release/Cargo.toml [hunk 5]
```diff
@@ -14,10 +14,11 @@ targets = ["x86_64-unknown-linux-gnu"]
...
-glob = "0.2"
+git2 = "0.17"
+glob = "0.3"
 tar = "0.4"
 tempfile = "3"
-toml = "0.4"
+toml_edit = "0.19"
+itertools = "0.10"
 
 [workspace]
```
Comment (ggwpez) [inline scripts/ci/node-template-release/Cargo.toml:24]: Line removed in https://github.com/paritytech/substrate/pull/14103

**Review by ggwpez — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate-macos
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2821353

**Pull request merged**